runtime/spark-tests/src/intTest/java/org/apache/polaris/service/spark/it/CatalogFederationIT.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.spark.it;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+import java.util.Map;
+import org.apache.polaris.service.it.test.CatalogFederationIntegrationTest;
+
+@TestProfile(CatalogFederationIT.CatalogFederationProfile.class)
+@QuarkusIntegrationTest
+public class CatalogFederationIT extends CatalogFederationIntegrationTest {
+
+  public static class CatalogFederationProfile implements QuarkusTestProfile {
+    @Override
+    public Map<String, String> getConfigOverrides() {
+      return Map.of(
+          "polaris.features.\"ENABLE_CATALOG_FEDERATION\"", "true",
+          "polaris.features.\"ALLOW_OVERLAPPING_CATALOG_URLS\"", "true");
```

Comment:
Would it be possible to use 2 different base locations + 2 different storage configs for the two catalogs, and thus avoid setting this?

Reply:
We need to use the same base location because we are using a single Polaris instance. So unless we have two different Polaris instances (which we can't do in the current integration test setup), we need this configuration.

Reply:
@poojanilangekar : Do you mean that federated catalogs check for location overlaps with internal catalogs?

Reply:
As @eric-maynard noted, it does look like federated catalogs should _not_ have a location define in Polaris. Is it possible to avoid declaring locations in federated catalogs now?

Reply:
Not at the moment. Making this change would require changing the API spec and making subsequent changes in the CreateCatalog validations.

Reply:
Would you be able to open an issue for sorting this out later?

Reply:
Done: #2356

Reply:
thx :+1:

Reply:
This PR is good to merge from my POV :+1: